Team417_CENTERSTAGE/src/main/java/org/firstinspires/ftc/team417_CENTERSTAGE/pidf/SimplePIDF.java

```diff
@@ -0,0 +1,78 @@
+package org.firstinspires.ftc.team417_CENTERSTAGE.pidf;
+
+
+public class SimplePIDF {
+    private double P;
+    private double I;
+    private double D;
+    private double F;
+
+    private double maxI = 1;
+    private double minI = -1;
+    private double maxOutput = 1;
+    private double minOutput = -1;
+    private boolean continuous = false;
+
+    private double error = 0;
+    private double integral = 0;
+    private double derivative = 0;
+    private double feedforward = 0;
+
+    private double lastInput = Double.NaN;
+
+    public SimplePIDF(double P, double I, double D, double F) {
+        this.P = P;
+        this.I = I;
+        this.D = D;
+        this.F = F;
+    }
+
+    public double calculate(double targetPosition) {
+        double currentPosition = getCurrentPosition();
+        error = targetPosition - currentPosition;
+
+        // Integral windup prevention
+        if (continuous) {
+            if (Math.abs(error) > (maxI - minI) / 2) {
+                error = (error > 0) ? error - maxI + minI : error + maxI - minI;
+            }
+        }
+
+        double dt = 0.02; // Assume a constant time step for simplicity
+        integral += error * dt;
+
+        derivative = (lastInput == Double.NaN) ? 0 : (currentPosition - lastInput) / dt;
```

Comment:
That NaN comparison is always false. The check should actually be done like this:

    Double.isNaN(lastInput)

(I learned this by looking at the warnings.)